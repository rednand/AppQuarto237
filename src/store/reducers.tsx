const initialState = {
  movies: [],
};

const appReducer = (state = initialState, action) => {
  return state;
};

export default appReducer;