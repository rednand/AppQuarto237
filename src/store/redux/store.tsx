import { createStore, applyMiddleware, combineReducers } from 'redux';
import thunk from 'redux-thunk';
import todoReducer from './reducers';
import { reducerSection } from './reducers';

const rootReducer = combineReducers({
  movies: todoReducer,
  //   movieSection: reducerSection,
});

const store = createStore(rootReducer, applyMiddleware(thunk));

export default store;
