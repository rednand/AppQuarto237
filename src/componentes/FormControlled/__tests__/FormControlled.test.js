import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import {
  toBeInTheDocument,
  toHaveLength,
  toHaveClass,
} from '@testing-library/jest-dom';
import UserForm from '..';
import store from './../../../store/redux/store';
import { Provider } from 'react-redux';
import userEvent from '@testing-library/user-event';

describe('Deve renderizar um campo de input', () => {
  test('tipo input', () => {
    render(
      <Provider store={store}>
        <UserForm />
      </Provider>,
    );
    const campoTexto = screen.getByPlaceholderText('Título');
    expect(campoTexto).toHaveAttribute('type', 'text');
  });

  test('input nome', () => {
    render(
      <Provider store={store}>
        <UserForm />
      </Provider>,
    );
    const campoTexto = screen.getByPlaceholderText('Título');
    expect(campoTexto).toBeInTheDocument();
  });

  test('input nome', async () => {
    render(
      <Provider store={store}>
        <UserForm />
      </Provider>,
    );

    const campoTexto = screen.getByPlaceholderText('Título');
    userEvent.type(campoTexto, 'teste');

    await waitFor(() => {
      expect(campoTexto).toHaveValue('teste');
    });
    expect(campoTexto).toMatchSnapshot();
  });

  test('input detalhes', async () => {
    render(
      <Provider store={store}>
        <UserForm />
      </Provider>,
    );

    const campoTexto = screen.getByPlaceholderText('Detalhes');
    userEvent.type(campoTexto, 'DetalhesDetalhesDetalhes');

    await waitFor(() => {
      expect(campoTexto).toHaveValue('DetalhesDetalhesDetalhes');
    });
    expect(campoTexto).toMatchSnapshot();
  });
});
