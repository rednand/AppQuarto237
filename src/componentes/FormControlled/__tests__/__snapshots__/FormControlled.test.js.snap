// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar um campo de input  1`] = `
<input
  class="sc-aXZVg iUtDJs form-control"
  name="name"
  placeholder="TÃ­tulo"
  style="color: rgb(28, 28, 28);"
  type="text"
  value="teste"
/>
`;

exports[`Deve renderizar um campo de input  2`] = `
<input
  class="sc-aXZVg iUtDJs form-control"
  name="overview"
  placeholder="Detalhes"
  type="text"
  value="DetalhesDetalhesDetalhes"
/>
`;
